movie-project/src/components/genres.js

```diff
@@ -0,0 +1,63 @@
+import { Form, Nav } from "react-bootstrap";
+import React, { useState, useEffect } from "react";
+
+export default function GenreSelector(props) {
+  const [popularMovies, setPopularMovies] = useState([]);
+  fetchMovieGenres(props.isLoaded, props.handleGenresList, props.setIsLoaded);
+  fetchPopularMovies(props.isLoaded, props.handleMovies, setPopularMovies);
+  let genresList = [...props.genresList];
+  return (
+    <Nav className="mr-auto mb-0">
+      <Form.Group className="mb-0" controlId="exampleForm.ControlSelect1">
+        <Form.Control
+          as="select"
+          onChange={e => {
+            let selectedGenre = Number(e.target.value);
+            const filteredMovies = popularMovies.filter(movie =>
+              movie.genre_ids.includes(selectedGenre)
+            );
+            props.handleMovies(filteredMovies);
+          }}
+        >
+          {genresList.map(genre => {
+            return (
+              <option key={genre.id} value={genre.id}>
+                {genre.name}
+              </option>
+            );
+          })}
+        </Form.Control>
+      </Form.Group>
+    </Nav>
+  );
+}
+
+function fetchMovieGenres(isLoaded, handleGenresList, setIsLoaded) {
+  if (!isLoaded) {
```

Comment:
don't use the `isLoaded` here and instead use the `useEffect` hook and the fetch inside it